### PR TITLE
guestfs: solve flakiness for guestfs tests

### PR DESCRIPTION
**Pull request by alicefr**
Sometimes, the libguestfs pod is deleted by the test clean-up but the function runGuestfsOnPVC hasn't gracefully terminated yet. The function runGuestfsOnPVC has a goroutine that returns when the channel of the fakeCreateAttacher is closed. Before this fix, there was a race between the `defer f.closeChannel()` and the test clean-up
Without the coordination, the [runGuestfsOnPVC](https://github.com/kubevirt/kubevirt/blob/main/tests/storage/guestfs.go#L85) function can return an error that the pod isn't existing anymore.  

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/storage/guestfs.go [hunk 1]
```diff
@@ -119,6 +121,21 @@ var _ = SIGDescribe("[rfe_id:6364][[Serial]Guestfs", func() {
 
 	}
 
+	waitForPodToBeDeletedOrTerminated := func(podName string) {
```
Comment (maya-r) [inline tests/storage/guestfs.go:124]: I think typically, for test speedup, instead of waiting for things to be deleted and cleaning up the environment between tests we prefer using unique names. What do you think about not waiting for the pod to be deleted or terminated at all and removing the AfterEach?

Comment (alicefr) [reply tests/storage/guestfs.go:124]: This is for synchronizing with the function [runGuestfsOnPVC](https://github.com/kubevirt/kubevirt/blob/main/tests/storage/guestfs.go#L85). If the pod is cleaned-up before this function terminates, we might have a failure in the AfterEach complaining that the pod `libguestfs-tool-pvc-fail-to-run-twice` doesn't exist anymore for example.
Please note, I'm not attending that the pod is fully deleted but in `Terminating` state. This means that the channel has already been closed.
Basically, there is a race between each `defer f.closeChannel()` and the test clean-up

File: tests/storage/guestfs.go [hunk 2]
```diff
@@ -200,16 +225,16 @@ var _ = SIGDescribe("[rfe_id:6364][[Serial]Guestfs", func() {
 			Expect(err).ToNot(HaveOccurred())
 			setGroup = false
 		})
-
 		AfterEach(func() {
-			err := virtClient.CoreV1().PersistentVolumeClaims(util.NamespaceTestDefault).Delete(context.Background(), pvcClaim, metav1.DeleteOptions{})
-			Expect(err).ToNot(HaveOccurred())
+			podName = libguestsTools + pvcClaim
```
Comment (maya-r) [inline tests/storage/guestfs.go:229]: podName is set twice both by the test and the aftereach, this might lead to future confusion

**Comment by maya-r** [comment]
I wonder if we can simplify this with having BeforeEach do createFakeAttacher and having AfterEach run f.closeChannel(), rather than relying on defer which has unclear ordering relative to ginkgo specific ways of running things after the test is completed.
(but I'm not 100% sure that AfterEach runs before namespace cleanup).

**Comment by alicefr** [comment]
@maya-r many thanks for the review! I added your suggestion

**Comment by maya-r** [comment]
/lgtm
(Although that will go away once you rebase)

**Comment by alicefr** [comment]
@maya-r I had to change the test setup because of some feedback from @rmohr. Would you mind give a second look to this?
Now, I added a `select` between the execution of the guestfs command and the done channel https://github.com/kubevirt/kubevirt/pull/8753/commits/6e87ffcccaad6d38e60934bfc07752a045f2a506 . In this way, we can gracefully close the guestfs goroutine

**Comment by rmohr** [comment]
> @maya-r I had to change the test setup because of some feedback from @rmohr. Would you mind giving a second look at this?
> Now, I added a `select` between the execution of the guestfs command and the done channel [6e87ffc](https://github.com/kubevirt/kubevirt/commit/6e87ffcccaad6d38e60934bfc07752a045f2a506) . In this way, we can gracefully close the guestfs goroutine

@alicefr I think that makes more sense since you deal more direct with the actual problem. If @maya-r is happy with it too, I can approve. Thanks for trying.

**Comment by rmohr** [comment]
/approve

approach works for me. @maya-r would be happy if you could have another look on the details.
